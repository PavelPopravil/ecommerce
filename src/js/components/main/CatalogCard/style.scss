@import "../../../../scss/main/variables";

.catalog-card {
  flex: 1;
  max-width: 100%;
  padding: 25px 20px;
  box-shadow: $b-shadow;
  text-decoration: none;

  &:hover,
  &:focus {
    box-shadow: $b-shadow-hover;
  }

  &__pic-wrap {
    height: 180px;
    margin-bottom: 15px;
  }

  &__pic {
    margin: 0 auto;
    max-height: 100%;
  }

  &__text-wrap {}

  &__title {}

  &__price {
    font-size: 16px;
    color: $link-color;
    font-family: $font-m;
  }
}