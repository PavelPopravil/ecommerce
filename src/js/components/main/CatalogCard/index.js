import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';

import './style.scss';

class CatalogCard extends React.PureComponent {

  static propTypes = {
    'pic': PropTypes.string.isRequired,
    'title': PropTypes.string.isRequired,
    'price': PropTypes.string.isRequired
  };

  render() {
    console.log('render CatalogCard');
    const {pic, title, price} = this.props;

    return (
      <Link className='catalog-card' to={'/about'}>
        <div className='catalog-card__pic-wrap'>
          <img className='catalog-card__pic' src={`${process.env.PUBLIC_URL}/img/${pic}`} alt={title} />
        </div>
        <div className='catalog-card__text-wrap'>
          <div className="h3 catalog-card__title">{title}</div>
          <div className="catalog-card__price">{price}</div>
        </div>
      </Link>
    )
  }
}

export default CatalogCard;