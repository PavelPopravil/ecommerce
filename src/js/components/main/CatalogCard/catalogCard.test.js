import React from 'react';
import {shallow} from 'enzyme';

import CatalogCard from './index';

describe('CatalogCard render', () => {

  const props = {
    path: 'mac',
    pic: 'catalog_imac.png',
    title: 'Заголовок карточки',
    price: 'Цена карточки'
  };

  const catalogCard = shallow(<CatalogCard {...props}/>);

  it('should render properly', () => {
    expect(catalogCard).toMatchSnapshot();
  });

  it('should contain picture', () => {
    expect(catalogCard.find('.catalog-card__pic')).toHaveLength(1);
  });

  it('should contain title', () => {
    const title = catalogCard.find('.catalog-card__title');
    expect(title).toHaveLength(1);
    expect(title.text()).toEqual(props.title)
  });

  it('should contain price', () => {
    const price = catalogCard.find('.catalog-card__price');
    expect(price).toHaveLength(1);
    expect(price.text()).toEqual(props.price);
  });

});