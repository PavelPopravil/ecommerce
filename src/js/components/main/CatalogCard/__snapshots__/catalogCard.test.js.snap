// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogCard render should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CatalogCard
    id={1}
    pic="catalog_imac.png"
    price="Цена карточки"
    title="Заголовок карточки"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="catalog-card__pic-wrap"
        >
          <img
            alt="Заголовок карточки"
            className="catalog-card__pic"
            src="/img/catalog_imac.png"
          />
        </div>,
        <div
          className="catalog-card__text-wrap"
        >
          <div
            className="h3 catalog-card__title"
          >
            Заголовок карточки
          </div>
          <div
            className="catalog-card__price"
          >
            Цена карточки
          </div>
        </div>,
      ],
      "className": "catalog-card",
      "replace": false,
      "to": "/product_list/1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Заголовок карточки"
            className="catalog-card__pic"
            src="/img/catalog_imac.png"
          />,
          "className": "catalog-card__pic-wrap",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Заголовок карточки",
            "className": "catalog-card__pic",
            "src": "/img/catalog_imac.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="h3 catalog-card__title"
            >
              Заголовок карточки
            </div>,
            <div
              className="catalog-card__price"
            >
              Цена карточки
            </div>,
          ],
          "className": "catalog-card__text-wrap",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Заголовок карточки",
              "className": "h3 catalog-card__title",
            },
            "ref": null,
            "rendered": "Заголовок карточки",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Цена карточки",
              "className": "catalog-card__price",
            },
            "ref": null,
            "rendered": "Цена карточки",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="catalog-card__pic-wrap"
          >
            <img
              alt="Заголовок карточки"
              className="catalog-card__pic"
              src="/img/catalog_imac.png"
            />
          </div>,
          <div
            className="catalog-card__text-wrap"
          >
            <div
              className="h3 catalog-card__title"
            >
              Заголовок карточки
            </div>
            <div
              className="catalog-card__price"
            >
              Цена карточки
            </div>
          </div>,
        ],
        "className": "catalog-card",
        "replace": false,
        "to": "/product_list/1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Заголовок карточки"
              className="catalog-card__pic"
              src="/img/catalog_imac.png"
            />,
            "className": "catalog-card__pic-wrap",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Заголовок карточки",
              "className": "catalog-card__pic",
              "src": "/img/catalog_imac.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="h3 catalog-card__title"
              >
                Заголовок карточки
              </div>,
              <div
                className="catalog-card__price"
              >
                Цена карточки
              </div>,
            ],
            "className": "catalog-card__text-wrap",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Заголовок карточки",
                "className": "h3 catalog-card__title",
              },
              "ref": null,
              "rendered": "Заголовок карточки",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Цена карточки",
                "className": "catalog-card__price",
              },
              "ref": null,
              "rendered": "Цена карточки",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
